### PR TITLE
Changes for federative requests

### PR DESCRIPTION
**Pull request by rjzondervan**

### CODE REVIEW

File: api/migrations/Version20240311121033.php [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('ALTER TABLE action ALTER id TYPE UUID');
```
Comment (WilcoLouwerse) [inline api/migrations/Version20240311121033.php]: Please remove uuid stuff from migrations

File: api/migrations/Version20240311125549.php [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('ALTER TABLE action ALTER id TYPE UUID');
```
Comment (WilcoLouwerse) [inline api/migrations/Version20240311125549.php]: Please remove uuid stuff from migrations

File: api/migrations/Version20240311132843.php [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('CREATE TABLE endpoint_gateway (endpoint_id UUID NOT NULL, gateway_id UUID NOT NULL, PRIMARY KEY(endpoint_id, gateway_id))');
+        $this->addSql('CREATE INDEX IDX_39C1245121AF7E36 ON endpoint_gateway (endpoint_id)');
+        $this->addSql('CREATE INDEX IDX_39C12451577F8E00 ON endpoint_gateway (gateway_id)');
+        $this->addSql('ALTER TABLE endpoint_gateway ADD CONSTRAINT FK_39C1245121AF7E36 FOREIGN KEY (endpoint_id) REFERENCES endpoint (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE');
+        $this->addSql('ALTER TABLE endpoint_gateway ADD CONSTRAINT FK_39C12451577F8E00 FOREIGN KEY (gateway_id) REFERENCES gateway (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE');
+        $this->addSql('ALTER TABLE action ALTER id TYPE UUID');
```
Comment (WilcoLouwerse) [inline api/migrations/Version20240311132843.php]: Please remove uuid stuff from migrations

File: api/src/Entity/Endpoint.php [hunk 4]
```diff
@@ -340,6 +340,12 @@ class Endpoint
      */
     private string $version = '0.0.0';
 
+    /**
+     * @Groups({"read", "write"})
+     * @ORM\ManyToMany(targetEntity=Gateway::class, inversedBy="federationEndpoints")
+     */
+    private $federationProxies;
```
Comment (WilcoLouwerse) [inline api/src/Entity/Endpoint.php]: Do we want a default value here? maybe null / empty array?

And what about @var & @MaxDepth 🤔

File: api/src/Entity/Gateway.php [hunk 5]
```diff
@@ -767,6 +767,12 @@ class Gateway
      */
     private $proxies;
 
+    /**
+     * @Groups({"read", "write"})
+     * @ORM\ManyToMany(targetEntity=Endpoint::class, mappedBy="federationProxies")
+     */
+    private $federationEndpoints;
```
Comment (WilcoLouwerse) [inline api/src/Entity/Gateway.php]: Do we want a default value here? maybe null / empty array?

And what about @var & @MaxDepth 🤔

**Review by WilcoLouwerse — CHANGES_REQUESTED** [review]

File: api/src/Entity/Endpoint.php (hunk 4 above)
Comment (rjzondervan) [reply api/src/Entity/Endpoint.php]: Default value is set by the constructor.
Will add documentation in the morning

**Review by WilcoLouwerse — APPROVED** [review]

**Review by bbrands02 — APPROVED** [review]

**Pull request merged**